File: 1034-subarrays-with-k-different-integers/subarrays-with-k-different-integers.java
class Solution {
    public static int find(int []nums,int k){
        HashMap<Integer,Integer>mp=new HashMap<>();
        int l=0;
        int r=0;
        int count=0;
    while( r<nums.length){
        mp.put(nums[r],mp.getOrDefault(nums[r],0)+1);
        while(l<=r && mp.size()>k){
            mp.put(nums[l],mp.get(nums[l])-1);
       
            if(mp.get(nums[l])==0){
                mp.remove(nums[l]);
            }
      
            l++;
            }
       count+=r-l+1;
        r++;
    }
    return count;

    }
    public int subarraysWithKDistinct(int[] nums, int k) {
        return find(nums,k)-find(nums,k-1);
        
    }
}

File: 1034-subarrays-with-k-different-integers/subarrays-with-k-different-integers.cpp
class Solution {
public:
int find(vector<int>&nums,int k){
    int left=0;
    int right=0;
    map<int,int>mp;
    int count1=0;
    while(right<nums.size()){
        mp[nums[right]]++;
        while(left<=right && mp.size()>k){
            mp[nums[left]]--;
            if(mp[nums[left]]==0)mp.erase(nums[left]);
            left++;
        }
       
            count1+=(right-left+1);
        
        right++;

    }
    return count1;
}
    int subarraysWithKDistinct(vector<int>& nums, int k) {
        return find(nums,k)-find(nums,k-1);
        
    }
};